src/cider/nrepl/middleware/spec.clj

```diff
@@ -0,0 +1,129 @@
+(ns cider.nrepl.middleware.spec
+  (:require [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
+            [cider.nrepl.middleware.util.spec :as spec-utils]
+            [clojure.tools.nrepl
+             [middleware :refer [set-descriptor!]]
+             [misc :refer [response-for]]]
+            [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.string :as str]
+            [clojure.walk :as walk]
+            [clojure.pprint :as pp]
+            [clojure.string :as str]))
+
+(defn str-non-colls
+  "Given a form, convert all non collection childs to str"
+  [form]
+  (walk/postwalk #(if (coll? %)
+                    %
+                    (str %))
+                 form))
+
+(defn spec-list
+  "Retrieves a list of all specs in the registry, sorted by ns/name.
+  If filter-prefix is not empty, keep only the specs with that prefix."
+  [filter-prefix]
```

Comment:
The `filter-prefix` should probably be a regexp instead, so it's maximally general.  That way it's more likely to fit the use-case the next guy comes up with without requiring a breaking change / new function.